polkadot/node/tracking-allocator/src/lib.rs

```diff
@@ -72,8 +72,10 @@ impl<T> Spinlock<T> {
 		}
 	}
 
+	// SAFETY: It should be only called from the guard's destructor. Calling it explicitly while
+	// the guard is alive is undefined behavior.
```

Comment:
Would be nice to explain why it's UB. After all, it wasn't obvious until we saw koute's example which segfaults.